### PR TITLE
WD-10729 - update contact form

### PR DESCRIPTION
**Pull request by lizzochek**
## Done

Updated contact form according to copy doc

## QA

- [demo link](https://canonical-com-1270.demos.haus/contact-us)
- [copy doc](https://docs.google.com/document/d/1m26cLDsx5OoCBFSblGgEUiGHsRtdLWAI6GBh-KnRIq0/edit)
- View the site on demo
- Run through the following [QA steps](https://discourse.canonical.com/t/qa-steps/152)
- Check that the forms are correct according to copy doc

## Issue / Card
[WD-10729](https://warthogs.atlassian.net/browse/WD-10729)

[WD-10729]: https://warthogs.atlassian.net/browse/WD-10729?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by lyubomir-popov** [comment]
The page background is white, whereas checkbox backgrounds - paper. We should stick with one background colour. If this is on a rebranded page, then the paper colour would be the correct one.

**Comment by lyubomir-popov** [comment]
I'd also move the asterisks at theend as they indent the text:
image

**Comment by lyubomir-popov** [comment]
Still not aligned with the underlying gid - the container should be 80 rem, with padding defined on 3 breakpoints, same asa with the `.row`:
image

**Comment by lyubomir-popov** [comment]
Fine above 1280px:
image

Still has some misalignments under that screen width: 
image
+1 lets fix the rest on the vanilla side
I've filed a vanilla enhancment issue here https://github.com/canonical/vanilla-framework/issues/5140

File: templates/partners/partial/_homepage-form.html [hunk 1]
```diff
@@ -1,197 +1,571 @@
...
+        </div>
+        <div class="col">
+          <div class="row u-no-padding">
+            <div class="col-4 u-sv3">
+              <strong>LTS within standard support</strong>
+              <label class="p-checkbox">
+                <input type="checkbox"
+                       aria-labelledby="22-04"
+                       class="p-checkbox__input"
+                       value="22.04 LTS" />
```
Comment (akbarkz) [inline templates/partners/partial/_homepage-form.html:44]: please add 24.04 LTS to the list of supported releases

File: templates/partners/partial/_homepage-form.html [hunk 2]
```diff
@@ -1,197 +1,571 @@
...
+              </label>
+              <label class="p-checkbox">
+                <input type="checkbox"
+                       aria-labelledby="github-upstream"
+                       class="p-checkbox__input /"
+                       value="Github/Upstream" />
+                <span class="p-checkbox__label" id="github-upstream">Github/Upstream</span>
+              </label>
+            </div>
+            <div class="col-4 u-sv3">
```
Comment (akbarkz) [inline templates/partners/partial/_homepage-form.html:257]: why are there two groups in the options?

File: templates/partners/partial/_homepage-form.html [hunk 3]
```diff
@@ -1,197 +1,571 @@
...
+              </label>
+              <label class="p-checkbox">
+                <input type="checkbox"
+                       aria-labelledby="fips-140"
+                       class="p-checkbox__input"
+                       value="FIPS 140" />
+                <span class="p-checkbox__label" id="fips-140">FIPS 140</span>
+              </label>
+            </div>
+            <div class="col-4 u-sv3">
```
Comment (akbarkz) [inline templates/partners/partial/_homepage-form.html:315]: same here, why do you have two groups of options?

File: templates/partners/partial/_homepage-form.html [hunk 4]
```diff
@@ -1,197 +1,571 @@
...
+              </label>
+              <label class="p-checkbox">
+                <input type="checkbox"
+                       aria-labelledby="project-team"
+                       class="p-checkbox__input /"
+                       value="The project team" />
+                <span class="p-checkbox__label" id="project-team">The project team</span>
+              </label>
+            </div>
+            <div class="col-4 u-sv3">
```
Comment (akbarkz) [inline templates/partners/partial/_homepage-form.html:375]: here too

**Review by akbarkz — CHANGES_REQUESTED** [review]

**Comment by akbarkz** [comment]
@lizzochek Some of the fields did not go through, take a look at the screenshot (I filled all the fields, i.e. selected at least one checkbox in each):
Screenshot 2024-06-14 at 16 57 38

**Comment by akbarkz** [comment]
@lizzochek In the screenshot that I provided above take a look at a question "How do you consume open source?:". Its answer is a dot, even though it should be a value of checkboxes that I checked. Same is for the question about "Who is responsible for tracking..."

File: static/js/modals.js [hunk 5]
```diff
@@ -147,3 +147,52 @@
...
+  const data = `Tell us about your project: ${document.getElementById("about-your-project")?.value}.\n
+    If you use Ubuntu, which version(s) are you using?: ${getCheckboxItemsAsCSV(ubuntuVersionsFieldset)}.\n
+    What kind of device are you using?: ${getCheckboxItemsAsCSV(deviceUsedFieldset)}.\n
+    How many devices?: ${getRadioItemValue(howManyMachinesFieldset)}.\n
+    How do you consume open source?: ${getCheckboxItemsAsCSV(consumeOpenSourceFieldset)}.\n
+    Do you have specific compliance or hardening requirements?: ${getCheckboxItemsAsCSV(hardeningRequirementsFieldset)}.\n
+    Who is responsible for tracking, testing and applying CVE patches in a timely manner?: ${getCheckboxItemsAsCSV(responsibleForTracking)}.\n
+    What advice are you looking for: ${document.getElementById("advice")?.value}.\n
+  `
+  const inputs = howManyMachinesFieldset?.querySelectorAll("input[name='how-many-machines-do-you-have']")
```
Comment (akbarkz) [inline static/js/modals.js:191]: let's be consistent, we either put semicolon after each JS line or don't put them at all.

File: static/sass/_pattern_forms.scss [hunk 6]
```diff
@@ -47,3 +47,13 @@
     }
   }
 }
+
+.is-wide-modal {
+  max-width: calc($grid-max-width + 2 * $spv--large - 2 * map-get($grid-margin-widths, small)) !important;
+
+  @media (min-width: $threshold-4-6-col) {
+  max-width: calc($grid-max-width + 2 * $spv--large - 2 * map-get($grid-margin-widths, default)) !important;
```
Comment (akbarkz) [inline static/sass/_pattern_forms.scss:55]: add indentation here

**Review by akbarkz — APPROVED** [review]

**Pull request merged**